client/src/components/profile/components/TimeLine.js

```diff
@@ -168,10 +144,38 @@ class Timeline extends Component {
   }
 }
 
+TimelineInner.propTypes = propTypes;
+
+function useIdToNameMap() {
+  const {
+    allChallengeNode: { edges }
+  } = useStaticQuery(graphql`
+    query challengeNodes {
+      allChallengeNode {
+        edges {
+          node {
+            fields {
+              slug
+            }
+            id
+            title
+          }
+        }
+      }
+    }
+  `);
+  const idToNameMap = new Map();
+  edges.forEach(({ node: { id, title, fields: { slug } } }) => {
+    idToNameMap.set(id, { challengeTitle: title, challengePath: slug });
+  });
+  return idToNameMap;
+}
+
+const Timeline = props => {
+  const idToNameMap = useIdToNameMap();
```

Comment:
```suggestion
```
Might as well just invoke the function on the next line, since we only use the variable once.